### DISCUSSION TITLE
No code action with neovim-lspconfig

### DISCUSSION THREAD

**Discussion by volgar1x** [python-lsp/python-lsp-server#182] [Q&A]
Hello everyone,

I recently decided to give pylsp a try after months of using pyright. The completions are very much more sane so thanks!

I still have some issues to make code actions work with Neovim though. Here is my current configuration,

```lua
  local nvim_lsp = require'lspconfig'
  local lsp_util = require'lspconfig/util'
  local lsp_signature = require'lsp_signature'

  local on_attach = function(client, bufnr)
    lsp_signature.on_attach()
    vim.api.nvim_buf_set_option(bufnr, 'omnifunc', 'v:lua.vim.lsp.omnifunc')
  end

  nvim_lsp.pylsp.setup {
    on_attach = on_attach,
    settings = {
      pylsp = {
        plugins = {
          mypy = { enabled = true },
          pylint = {
            enabled = true,
            args = { "--rcfile=pylint.ini" },
          },
          rope_completion = { enabled = true },
          black = { enabled = true },
        },
      },
    },
    flags = {
      debounce_text_changes = 150,
    },
  }
```

And also, do you know how I can enable auto-import?

**Comment by bagel897** [comment]
1. Install rope, and possibly pylsp-rope using pip / system package manager. 
2. Pass capabilities to pylsp
3. for autoimport, wait for [me](https://github.com/python-rope/rope/pull/464) (or help!)
4. Tip: remove mypy and black from your configs, and instead install their respective plugins. They only work when installed.
